File: languages/go/entity/availability/hosted_loader.go
package availability

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// HostedLoader represents a loader for hosted data availability.
type HostedLoader struct {
	Type proto.DataAvailabilityType
	Args LoaderArgs
}

// NewHostedLoader creates a HostedLoader instance with the provided identifier (ex: c137fded-cb04-4c6e-9415-1e7baf48b659).
func NewHostedLoader(id string) HostedLoader {
	return HostedLoader{
		Type: proto.DataAvailabilityType_HOSTED,
		Args: LoaderArgs{
			Id: id,
		},
	}
}

func (e HostedLoader) ToProto() *proto.Loader {
	return &proto.Loader{
		Type: e.Type,
		Args: e.Args.ToProto(),
	}
}

File: languages/js/src/entity/availability/hosted_loader.ts
import * as proto from "../../bridge/proto/bloock_availability_entities";
import { Loader } from "./loader";
import { LoaderArgs } from "./loader_args";

/**
 * Represents a loader for hosted data availability.
 */
export class HostedLoader implements Loader {
  type: proto.DataAvailabilityType;
  args: LoaderArgs;

  /**
   * Constructs a HostedLoader object with the specified parameters.
   * @param id
   */
  constructor(id: string) {
    this.type = proto.DataAvailabilityType.HOSTED;
    this.args = new LoaderArgs(id);
  }

  public toProto(): proto.Loader {
    return proto.Loader.fromPartial({
      type: this.type,
      args: this.args.toProto()
    });
  }
}
